3rdparty/arc4random-src/arc4random_uniform.cpp

```diff
@@ -0,0 +1,84 @@
+/*	$OpenBSD: arc4random_uniform.c,v 1.2 2015/09/13 08:31:47 guenther Exp $	*/
+
+/*
+ * Copyright (c) 2008, Damien Miller <djm@openbsd.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+// This is arc4random_uniform.c from OpenBSD's source tree at
+// $OPENBSD_ROOT/lib/libc/crypt/arc4random_uniform.c.
+//
+// Mumble-specific changes:
+//
+// - Change file extension to .cpp and rename to mumble_arc4random_uniform.cpp.
+// - Add CryptographicRandom.h include.
+// - Remove <sys/types.h> and <stdlib.h> includes, include <stdint.h> for uint32_t.
+// - Added an implementation of arc4random called mumble_arc4random.
+// - Rename arc4random_uniform to mumble_arc4random_uniform.
+// - Change call to arc4random in mumble_arc4random_uniform to call
+//   mumble_arc4random instead of arc4random.
+// - Remove DEF_WEAK for arc4random_uniform.
+// - Disable MSVC warning 4146.
+
+#include <stdint.h>
+#include "CryptographicRandom.h"
+
+#if defined(_MSC_VER)
+// Disable "unary minus operator applied to unsigned type, result still unsigned"
+// warning. The unary minus operator used in arc4random_uniform is perfectly valid.
+# pragma warning(disable: 4146)
```

Comment:
I guess I had bad memories of warnings being disabled in header files and bleeding elsewhere. The [pragma warning docs](https://msdn.microsoft.com/en-us/library/2c8f766e.aspx) are not really explicit. Other places talk of TU tho. So nvm.